lib/screens/landing/login_screen.dart

```diff
@@ -24,32 +39,107 @@ class _LoginScreenState extends State<LoginScreen> {
   }
 
   void _userLogin() async {
-    const String apiUrl = 'http://localhost:8080/users/login';
+    if (emailController.text.isEmpty) {
+      Fluttertoast.showToast(
+        msg: "Please enter your email or username",
+        toastLength: Toast.LENGTH_SHORT,
+        gravity: ToastGravity.CENTER,
+        timeInSecForIosWeb: 1,
+        backgroundColor: Colors.grey,
+        textColor: Colors.white,
+      );
+      return;
+    }
+
+    if (passwordController.text.isEmpty) {
+      Fluttertoast.showToast(
+        msg: "Please enter your password",
+        toastLength: Toast.LENGTH_SHORT,
+        gravity: ToastGravity.CENTER,
+        timeInSecForIosWeb: 1,
+        backgroundColor: Colors.grey,
+        textColor: Colors.white,
+      );
+      return;
+    }
+
+    setState(() {
+      isLoading = true;
+      searchResult = {};
+    });
+
+    const String apiUrl_email = 'http://localhost:8080/users/login';
+    const String apiUrl_username = 'http://localhost:8080/users/username/login';
 
     // Create a map with the collected data
-    final Map<String, dynamic> userData = {
+    final Map<String, dynamic> userData_email = {
       'userEmail': emailController.text,
       'userPassword': passwordController.text,
     };
-    print(userData);
+    final Map<String, dynamic> userData_username = {
+      'username': emailController.text,
+      'userPassword': passwordController.text,
+    };
+
     try {
-      final response = await http.post(
-        Uri.parse(apiUrl),
-        body: json.encode(userData),
+      final response_email = http.post(
+        Uri.parse(apiUrl_email),
+        body: json.encode(userData_email),
         headers: {'Content-Type': 'application/json'},
       );
```

Comment:
same here, if we're able to combine the logic on the backend for the user email and username to just one call, we would only have to send one http.post request to our backend which is more efficient